test/sorting.jl

```diff
@@ -721,10 +721,9 @@ end
     for alg in safe_algs
         @test sort(1:n, alg=alg, lt = (i,j) -> v[i]<=v[j]) == perm
     end
-    # This could easily break with minor heuristic adjustments
-    # because partialsort is not even guaranteed to be stable:
-    @test partialsort(1:n, 172, lt = (i,j) -> v[i]<=v[j]) == perm[172]
-    @test partialsort(1:n, 315:415, lt = (i,j) -> v[i]<=v[j]) == perm[315:415]
+    # Broken by the introduction of BracketedSort in #52006 which is unstable
+    @test_broken partialsort(1:n, 172, lt = (i,j) -> v[i]<=v[j]) == perm[172]
```

Comment:
Is this even worth having as a test if all we do is to flip it between `@test` and `@test_broken` depending on the current state of the implementation?

Reply:
Yes. If I had listened to the test failure instead of blindly flipping it in https://github.com/JuliaLang/julia/pull/52006 then it would have caught the bug that this PR fixes.

This should not be a false positive for changes elsewhere in the codebase so the labor of switching it back and forth should be minimal.

Reply:
But changing a `@test` to a `@test_broken` means you have introduced a bug. But if the current code does not have a bug, then this test should not exist.

Reply:
This is a unit test of an internal to help identify unintended behavioral changes. It's okay to have tests that are not testing something that is a part of the public API. The only problem I can see with this test is the potential for false positives. However, so far it has had one true positive and zero false positives so I'm not worried about that yet.